Resource/PYTHON/multilateration/multilateration.py

```diff
@@ -0,0 +1,287 @@
+
+# Initializing the variables maximum_time_diff and max_hydrophone_distance.
+# These values are updated in the function initialize_trilateration_globals
+
+# Initialized to -1 such that it's easy to check if the variables are incorrect
+
+max_hydrophone_distance = -1.0
+max_time_diff = -1.0
+
+# Functions for initializing
+
+
+def initialize_trilateration_globals():
+
+    # Calculating the distances between the hydrophones
+    dist_port_starboard = float(
+        math.sqrt(
+            math.pow(
+                param.HydrophoneDetails.STARBOARD_HYD_X
+                - param.HydrophoneDetails.PORT_HYD_X,
+                2,
+            )
+            + math.pow(
+                param.HydrophoneDetails.STARBOARD_HYD_Y
+                - param.HydrophoneDetails.PORT_HYD_Y,
+                2,
+            )
+            + math.pow(
+                param.HydrophoneDetails.STARBOARD_HYD_Z
+                - param.HydrophoneDetails.PORT_HYD_Z,
+                2,
+            )
+        )
+    )
+    dist_port_stern = float(
+        math.sqrt(
+            math.pow(
+                param.HydrophoneDetails.STERN_HYD_X
+                - param.HydrophoneDetails.PORT_HYD_X,
+                2,
+            )
+            + math.pow(
+                param.HydrophoneDetails.STERN_HYD_Y
+                - param.HydrophoneDetails.PORT_HYD_Y,
+                2,
+            )
+            + math.pow(
+                param.HydrophoneDetails.STERN_HYD_Z
+                - param.HydrophoneDetails.PORT_HYD_Z,
+                2,
+            )
+        )
+    )
+    dist_starboard_stern = float(
+        math.sqrt(
+            math.pow(
+                param.HydrophoneDetails.STERN_HYD_X
+                - param.HydrophoneDetails.STARBOARD_HYD_X,
+                2,
+            )
+            + math.pow(
+                param.HydrophoneDetails.STERN_HYD_Y
+                - param.HydrophoneDetails.STARBOARD_HYD_Y,
+                2,
+            )
+            + math.pow(
+                param.HydrophoneDetails.STERN_HYD_Z
+                - param.HydrophoneDetails.STARBOARD_HYD_Z,
+                2,
+            )
+        )
+    )
+
+    # Finding the maximum distance
+    max_hydrophone_distance = max(
+        dist_port_starboard, max(dist_starboard_stern, dist_port_stern)
+    )
+
+    # Calculating max time allowed over that distance
+    max_time_diff = (1 + param.SystemMargins.MARGIN_TIME_EPSILON) * (
+        max_hydrophone_distance / param.PhysicalConstants.SOUND_SPEED
+    )
+
+    # Returning if both variables have been set correctly
+    return check_initialized_globals()
+
+
+def initialize_A_matrix():  # return type Matrix_2_3_f
+    A = np.zeros((2, 3))
+    return A
+
+
+def initialize_B_vector():
+    B = np.zeros((2, 1))
+    return B
+
+
+# Functions to check if signals/data are valid
+
+
+def check_initialized_globals():  # return type uint8_t
+    return max_hydrophone_distance != -1 and max_time_diff != 1
```

Comment:
shouldn't this be
```suggestion
    return max_hydrophone_distance != -1 and max_time_diff != -1
```
?
Or if you use `None` to initialize them instead you can do
```python
return max_hydrophone_distance is not None and max_time_diff is not None
```
which IMO is a lot more readable.